replica/table.cc

```diff
@@ -1150,6 +1150,7 @@ future<bool> table::perform_offstrategy_compaction() {
 }
 
 future<> table::perform_cleanup_compaction(compaction::owned_ranges_ptr sorted_owned_ranges) {
+    co_await flush();
```

Comment:
Does flush() wait for the flush to complete? Not from my reading of the code.

Reply:
Why not?
table::flush -> _dirty_memory_manager->flush_one -> mtlist.seal_active_memtable -> table::seal_active_memtable

The latter waits on any previous flush using _flush_barrier.advance_and_await
and then waits on try_flush_memtable_to_sstable which awaits write_memtable_to_sstable
and if successful, calls update_cache and finally cg.memtables()->erase.

Reply:
It's the mechanism used for implementing the explicit flush API which waits for its completion.

Reply:
seal_active_memtable() also flushes the memtable and waits for it? It's terribly misnamed.

Reply:
> seal_active_memtable() also flushes the memtable and waits for it? It's terribly misnamed.

Yes, and I agree about the name.
It's been like that since practically forever: see 0eb842dc5bc340ccd79a24f191c8c3e72400d492

Reply:
Ah, the function should have been renamed in that commit.